src/main/kotlin/org/elaastic/questions/player/PlayerController.kt

```diff
@@ -213,22 +221,68 @@ class PlayerController(
             val teacher = user == sequence.owner
 
             return if (teacher)
-                playAssignmentForTeacher(user, model, sequence, httpServletRequest)
-            else playAssignmentForLearner(user, model, sequence, httpServletRequest)
+                playAssignmentForTeacher(
+                    user,
+                    model,
+                    sequence,
+                    openedPane ?: "assignments",
+                    httpServletRequest
+                )
+            else
+                playAssignmentForLearner(
+                    user,
+                    model,
+                    sequence,
+                    openedPane ?: "assignments",
+                    httpServletRequest
+                )
         }
 
     }
 
-
     private fun playAssignmentForTeacher(
         user: User,
         model: Model,
         sequence: Sequence,
+        openedPane: String,
         httpServletRequest: HttpServletRequest,
     ): String {
 
-        val assignment = sequence.assignment!!
-        val nbRegisteredUsers = assignmentService.getNbRegisteredUsers(assignment)
+        val assignment: Assignment = sequence.assignment!!
+        val previousSequence: Sequence? = sequenceService.findPreviousSequence(sequence)
+        val nextSequence: Sequence? = sequenceService.findNextSequence(sequence)
+        val registeredUsers: List<LearnerAssignment> = assignmentService.getRegisteredUsers(assignment)
+        val nbRegisteredUsers = registeredUsers.size
+        val responses: List<Response> = interactionService.findAllResponsesBySequenceOrderById(sequence)
+        val attendeesSequences: MutableMap<Long, ILearnerSequence> = mutableMapOf()
+
+        // Associate each learner with the number of evaluations he made
+        val evaluationCountByUser = registeredUsers.associateWith {
+            peerGradingService.countEvaluationsMadeByUser(it.learner, sequence)
```

Comment:
N+1 Select issue. I wonder if it is not possible to get all required informations with only one request fetching all registered users with attached responses and attached gradings for each responses.

Reply:
A solution can probably be found through Entity Graphs:

- https://www.baeldung.com/jpa-entity-graph
- https://www.baeldung.com/spring-data-jpa-named-entity-graphs

The are some usages in Elaastic:

- https://github.com/elaastic/elaastic-questions-server/blob/develop/src/main/kotlin/org/elaastic/questions/subject/Subject.kt
- https://github.com/elaastic/elaastic-questions-server/blob/develop/src/main/kotlin/org/elaastic/questions/subject/SubjectRepository.kt

Reply:
Fix in [6040614](https://github.com/elaastic/elaastic-questions-server/pull/247/commits/6040614ff6ad49074ce6d5813de86bbfc56141c8)